sdks/python/apache_beam/yaml/main.py

```diff
@@ -64,9 +72,19 @@ def _pipeline_spec_from_args(known_args):
   return pipeline_yaml
 
 
+class _BeamFileIOLoader(jinja2.BaseLoader):
+  def get_source(self, environment, path):
+    source = FileSystems.open(path).read().decode()
+    return source, path, lambda: True
+
+
 def run(argv=None):
   known_args, pipeline_args = _configure_parser(argv)
-  pipeline_yaml = _pipeline_spec_from_args(known_args)
+  pipeline_yaml = (  # keep formatting
+      jinja2.Environment(
+          undefined=jinja2.StrictUndefined, loader=_BeamFileIOLoader())
+      .from_string(_pipeline_spec_from_args(known_args))
+      .render(**known_args.jinja_variables or {}))
```

Comment:
Something I just thought of here - should we be rendering the original yaml or the formatted yaml in display data? Or maybe both? I'd probably vote for both

Reply:
Yes, I think both are useful.

Reply:
Done.